File: Software/StarPRNT SDK/ObjectiveC SDK/ObjectiveC SDK/Functions/AllReceiptsFunctions.m
//
//  AllReceiptsFunctions.m
//  ObjectiveC SDK
//
//  Created by Yuji on 2016/**/**.
//  Copyright © 2016年 Star Micronics. All rights reserved.
//

#import "AllReceiptsFunctions.h"

#import <SMCloudServices/SMCSAllReceipts.h>

@implementation AllReceiptsFunctions

+ (NSData *)createRasterReceiptData:(StarIoExtEmulation)emulation
                   localizeReceipts:(ILocalizeReceipts *)localizeReceipts
                            receipt:(BOOL)receipt
                               info:(BOOL)info
                             qrCode:(BOOL)qrCode
                         completion:(void (^)(NSInteger statusCode, NSError *error))completion {
    UIImage *image = [localizeReceipts createRasterReceiptImage];
    
    NSString *urlString = [SMCSAllReceipts uploadBitmap:image completion:completion];
    
    if (receipt == YES ||
        info    == YES ||
        qrCode  == YES) {
        ISCBBuilder *builder = [StarIoExt createCommandBuilder:emulation];
        
        [builder beginDocument];
        
        if (receipt == YES) {
            [builder appendBitmap:image diffusion:NO];
        }
        
        if (info   == YES ||
            qrCode == YES) {
            NSData *allReceiptsData = [SMCSAllReceipts generateAllReceipts:urlString emulation:emulation info:info qrCode:qrCode];
            
//          [builder appendData   :allReceiptsData];
            [builder appendRawData:allReceiptsData];
        }
        
        [builder appendCutPaper:SCBCutPaperActionPartialCutWithFeed];
        
        [builder endDocument];
        
        return [builder.commands copy];
    }
    
    return nil;
}

+ (NSData *)createScaleRasterReceiptData:(StarIoExtEmulation)emulation
                        localizeReceipts:(ILocalizeReceipts *)localizeReceipts
                                   width:(NSInteger)width
                               bothScale:(BOOL)bothScale
                                 receipt:(BOOL)receipt
                                    info:(BOOL)info
                                  qrCode:(BOOL)qrCode
                              completion:(void (^)(NSInteger statusCode, NSError *error))completion {
    UIImage *image = [localizeReceipts createScaleRasterReceiptImage];
    
    NSString *urlString = [SMCSAllReceipts uploadBitmap:image completion:completion];
    
    if (receipt == YES ||
        info    == YES ||
        qrCode  == YES) {
        ISCBBuilder *builder = [StarIoExt createCommandBuilder:emulation];
        
        [builder beginDocument];
        
        if (receipt == YES) {
            [builder appendBitmap:image diffusion:NO width:width bothScale:bothScale];
        }
        
        if (info   == YES ||
            qrCode == YES) {
            NSData *allReceiptsData = [SMCSAllReceipts generateAllReceipts:urlString emulation:emulation info:info qrCode:qrCode];
            
//          [builder appendData   :allReceiptsData];
            [builder appendRawData:allReceiptsData];
        }
        
        [builder appendCutPaper:SCBCutPaperActionPartialCutWithFeed];
        
        [builder endDocument];
        
        return [builder.commands copy];
    }
    
    return nil;
}

@end

File: Software/StarPRNT SDK/Swift SDK/Swift SDK/Functions/AllReceiptsFunctions.swift
//
//  AllReceiptsFunctions.swift
//  Swift SDK
//
//  Created by Yuji on 2016/**/**.
//  Copyright © 2016年 Star Micronics. All rights reserved.
//

import Foundation

class AllReceiptsFunctions {
    static func createRasterReceiptData(emulation: StarIoExtEmulation, localizeReceipts: ILocalizeReceipts, receipt: Bool, info: Bool, qrCode: Bool, completion: ((Int, NSError!) -> Void)) -> NSData? {
        let image: UIImage = localizeReceipts.createRasterReceiptImage()!
        
        let urlString: String = SMCSAllReceipts.uploadBitmap(image, completion: completion)
        
        if receipt == true ||
           info    == true ||
           qrCode  == true {
            let builder: ISCBBuilder = StarIoExt.createCommandBuilder(emulation)
            
            builder.beginDocument()
            
            if receipt == true {
                builder.appendBitmap(image, diffusion: false)
            }
            
            if info   == true ||
               qrCode == true {
                let allReceiptsData: NSData = SMCSAllReceipts.generateAllReceipts(urlString, emulation: emulation, info: info, qrCode: qrCode)
                
//              builder.appendData   (allReceiptsData)
                builder.appendRawData(allReceiptsData)
            }
            
            builder.appendCutPaper(SCBCutPaperAction.PartialCutWithFeed)
            
            builder.endDocument()
            
            return builder.commands.copy() as? NSData
        }
        
        return nil
    }
    
    static func createScaleRasterReceiptData(emulation: StarIoExtEmulation, localizeReceipts: ILocalizeReceipts, width: Int, bothScale: Bool, receipt: Bool, info: Bool, qrCode: Bool, completion: ((Int, NSError!) -> Void)) -> NSData? {
        let image: UIImage = localizeReceipts.createScaleRasterReceiptImage()!
        
        let urlString: String = SMCSAllReceipts.uploadBitmap(image, completion: completion)
        
        if receipt == true ||
           info    == true ||
           qrCode  == true {
            let builder: ISCBBuilder = StarIoExt.createCommandBuilder(emulation)
            
            builder.beginDocument()
            
            if receipt == true {
                builder.appendBitmap(image, diffusion: false, width: width, bothScale: bothScale)
            }
            
            if info   == true ||
               qrCode == true {
                let allReceiptsData: NSData = SMCSAllReceipts.generateAllReceipts(urlString, emulation: emulation, info: info, qrCode: qrCode)
                
//              builder.appendData   (allReceiptsData)
                builder.appendRawData(allReceiptsData)
            }
            
            builder.appendCutPaper(SCBCutPaperAction.PartialCutWithFeed)
            
            builder.endDocument()
            
            return builder.commands.copy() as? NSData
        }
        
        return nil
    }
}
